### ISSUE TITLE
macOS Minimum Version for ARM64 Builds

### ISSUE THREAD

**Issue by BryanHaley** [coop-deluxe/sm64coopdx#511]
What is the rationale behind setting 14 as the minimum version for macOS ARM64 builds? ARM64 Macs can use as early as version 12.0 and this app isn't using any feature that requires it to be >12.0.

**Comment by BryanHaley** [comment]
Simply setting it to 12 has it build and run fine on my ARM64 Mac on 13.5, though it does complain about liblua.a being built against a newer version of MacOS; that would need to get rebuilt for 12.0 (which should also be fine) to make a "proper" fix.

**Comment by EmeraldLoc** [comment]
This is a great question, and the only answers I have is because my Mac is on macOS 15, and rhe glew and sdl2 libs are on macOS 14, so it’s more about my system restraints than anything.

**Comment by BryanHaley** [comment]
Yeah this tends to be a pain about macOS development -- when brew gives you a library, you can't tell it "I want this built against xyz version of macOS". The typical solutions to this are:
- Build libraries from source where you can specify minimum macOS version (in this case I think only liblua is of concern? At least it didn't complain about other libs when linking)
- Get precompiled libraries from a system with the macOS version you want (can use virtual machine for this)
- Build the whole project on a system with the macOS version you want (can use virtual machine for this)

Another solution is github has [macOS 12 runner images](https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md) for github actions available; anything needed can be built in those and you'll get something that works versus macOS 12.

If it's really only liblua though, the easiest solution is probably just building that from source.
